File: SPCServer/springboot/src/main/java/com/lcaohoanq/shoppe/services/token/ITokenService.java
package com.lcaohoanq.shoppe.services.token;

import com.lcaohoanq.shoppe.exceptions.base.DataNotFoundException;
import com.lcaohoanq.shoppe.models.Token;
import com.lcaohoanq.shoppe.models.User;
import org.springframework.stereotype.Service;

@Service

public interface ITokenService {
    Token addToken(User user, String token, boolean isMobileDevice);
    Token refreshToken(String refreshToken, User user) throws Exception;
    void deleteToken(String token, User user) throws DataNotFoundException;
    Token findUserByToken(String token) throws DataNotFoundException;
    void setTokenExpired();
}

File: auth-service/src/main/kotlin/com/lcaohoanq/authservice/domains/token/ITokenService.kt
package com.lcaohoanq.authservice.domains.token

import com.lcaohoanq.authservice.domains.user.User
import com.lcaohoanq.common.dto.TokenPort

interface ITokenService {

    fun getAll(): List<TokenPort.TokenResponse>
    fun addToken(userId: Long, token: String): Token
    fun refreshToken(refreshToken: String, user: User): Token
    fun deleteToken(token: String, user: User)
    fun findUserByToken(token: String): Token

}